examples/todomvc/components/TodoItem.js

```diff
@@ -0,0 +1,69 @@
+import React, { Component, PropTypes } from 'react';
+import classnames from 'classnames';
+import TodoTextInput from './TodoTextInput';
+
+
+export default class TodoItem extends Component {
+  static propTypes = {
+    todo: PropTypes.object.isRequired,
+    editTodo: PropTypes.func.isRequired,
+    deleteTodo: PropTypes.func.isRequired,
+    markTodo: PropTypes.func.isRequired
+  };
+
+  constructor(props, context) {
+    super(props, context);
+    this.state = {
+      edit: false
+    };
+  }
+
+  onDoubleClick() {
+    this.setState({ edit: true });
+  }
+
+  onSave(id, text) {
+    if (text.length === 0) {
+      this.props.deleteTodo(id);
+    } else {
+      this.props.editTodo(id, text);
+    }
+    this.setState({ edit: false });
+  }
+
+  render() {
+    const {todo, markTodo, deleteTodo, editTodo} = this.props;
+
+    let view;
+
+    if (this.state.edit) {
+      view = (<TodoTextInput
+                  text={todo.text}
+                  edit={this.state.edit}
```

Comment:
`this.state.edit` is only true on double click. When the `onsave` function triggers it sets the `edit` to `false` again